url7f43a4752f_shalynnho_pokemonfactory_tgz-pJ8-Networking_ServerJ8/src/agent/CameraAgent.java

```diff
@@ -0,0 +1,554 @@
+					 * nests.get(i + 1).state == NestStatus.READY_TO_RE_PHOTOGRAPH || nests.get(i + 1).state ==
+					 * NestStatus.READY_TO_RE_PHOTOGRAPH_AGAIN)) {
+					 */
+					if (nests.get(i).state == NestStatus.READY
+							|| nests.get(i).state == NestStatus.READY_TO_RE_PHOTOGRAPH
+							|| nests.get(i).state == NestStatus.READY_TO_RE_PHOTOGRAPH_AGAIN) {
+						if (nests.get(i + 1).state == NestStatus.READY
+								|| nests.get(i + 1).state == NestStatus.READY_TO_RE_PHOTOGRAPH
+								|| nests.get(i + 1).state == NestStatus.READY_TO_RE_PHOTOGRAPH_AGAIN) {
+							print("Taking photos of nests");
+							takePictureOfNest(nests.get(i), nests.get(i + 1));
+							return true;
+						} else if (nests.get(i + 1).state == NestStatus.NOT_READY) {
+							print("Started a timer for nest " + (i + 1));
+							nests.get(i + 1).state = NestStatus.STARTED_TIMER;
+							final Nest tempNest = nests.get(i + 1).nest;
+							timer.schedule(new TimerTask() {
+								@Override
+								public void run() {
+									msgIAmFull(tempNest);
+								}
+							}, 1000);
+							return true;
+						}
+					} else if (nests.get(i + 1).state == NestStatus.READY
+							|| nests.get(i + 1).state == NestStatus.READY_TO_RE_PHOTOGRAPH
+							|| nests.get(i + 1).state == NestStatus.READY_TO_RE_PHOTOGRAPH_AGAIN) {
+						if (nests.get(i).state == NestStatus.NOT_READY) {
+							print("Started a timer for nest " + i);
+							nests.get(i).state = NestStatus.STARTED_TIMER;
+							final Nest tempNest = nests.get(i).nest;
+							timer.schedule(new TimerTask() {
+								@Override
+								public void run() {
+									msgIAmFull(tempNest);
+								}
+							}, 1000);
+							return true;
+						}
+					}
+					// }
+				}
+			}
+		}
+
+		synchronized (nests) {
+			for (MyNest n : nests) {
+				// print("nest "+n.nest.currentPartType+ " state is "+n.state);
+				if (n.state == NestStatus.PHOTOGRAPHED || n.state == NestStatus.RE_PHOTOGRAPHED_ONCE
+						|| n.state == NestStatus.RE_PHOTOGRAPHED_TWICE) {
+					tellPartsRobot(n);
+					return true;
+				}
+			}
+		}
+
+		if (mk != null) {
+			if (mk.ks == KitStatus.NOT_READY) {
+				takePictureOfKit(mk);
+				return true;
+			} else if (mk.ks == KitStatus.DONE) {
+				tellKitRobot(mk);
+				return true;
+			} else if (mk.ks == KitStatus.FAILED) {
+				tellKitRobotFailure(mk);
+				return true;
+			}
+
+		}
+
+		return false;
+	}
+
+	/*
+	 * Actions
+	 */
+	private void tellKitRobot(MyKit kit) {
+		mk = null;
+		kitRobot.msgKitPassedInspection();
+		stateChanged();
+	}
+
+	private void tellKitRobotFailure(MyKit kit) {
+		mk = null;
+		kitRobot.msgKitFailedInspection();
+		stateChanged();
+	}
+
+	private void takePictureOfKit(MyKit kit) {
+		mk.ks = KitStatus.PICTURE_BEING_TAKEN;
+		if (guiCamera != null) {
+			guiCamera.takeKitPhoto(kit.kit.kitGraphics);
+			print("Blocking");
+			try {
+				animation.acquire();
+			} catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)